Makefile

```diff
@@ -1,19 +1,12 @@
-gen:
-	@go generate ./...
-
-lint:
-	# These PATH hacks are temporary until prow properly sets its paths
-	@PATH=${PATH}:${GOPATH}/bin scripts/check_license.sh
-	@PATH=${PATH}:${GOPATH}/bin scripts/run_golangci.sh
-
-fmt:
-	@scripts/run_gofmt.sh
-
-test:
-	@go test -race ./...
-
-test_with_coverage:
-	@go test -race -coverprofile=coverage.txt -covermode=atomic ./...
-	@curl -s https://codecov.io/bash | bash -s -- -c -F aFlag -f coverage.txt
+img := gcr.io/istio-testing/build-tools:2019-08-02
+pwd := $(shell pwd)
+uid := $(shell id -u)
+gocache := $(shell go env GOCACHE)
+gomod := $(GOPATH)/pkg/mod
+run = docker run -t -i --sig-proxy=true -u $(uid) --rm -v /etc/passwd:/etc/passwd:ro -v $(gocache):/gocache -v $(gomod):/go/pkg/mod -v $(pwd):/work -w /work $(img)
```

Comment:
for this command - you might instead use ```run = docker run -t --sig-proxy=true -u $(uid) --rm \
+       -v /etc/passwd:/etc/passwd:ro \
+       --mount type=bind,source="$(gocache)",destination="/gocache",bind-propagation=rshared \
+       --mount type=bind,source="$(gomod)",destination="/go/pkg/mod",bind-propagation=rshared \
+       --mount type=bind,source="$(pwd)",destination="/work",bind-propagation=rshared \
+       -w /work $(img)
```

`-v` will create directories as root if they are not present.  This presents a problem in that the build will not proceed on a clean system.  `--mount` does not create directories and allows finer grained control over the mount operation.  It may be necessary to add some `mkdir -p` operations in here as well.